### ISSUE TITLE
2023.2_mpich_znver3_psmp.def does not work

### ISSUE THREAD

**Issue by iamid0** [cp2k/cp2k#2955]
Tried with the following cmd, 
```
apptainer build -B $PWD:/mnt cp2k-2023.2_mpich_znver3_psmp.sif 2023.2_mpich_znver3_psmp.def | tee cp2k-2023.2_mpich_znver3_psmp.log
```

Here is the error message, 

```
==================== Installing OpenBLAS ====================
OpenBLAS-0.3.23.tar.gz is found
Installing from scratch into /opt/cp2k/tools/toolchain/install/openblas-0.3.23
Installing OpenBLAS library for target ZNVER3
getarch.c:1698:2: error: #error "The TARGET specified on the command line or in Makefile.rule is not supported. Please choose a target from TargetList.txt"
 1698 | #error "The TARGET specified on the command line or in Makefile.rule is not supported. Please choose a target from TargetList.txt"
```

I made only one modification to the `2023.2_mpich_znver3_psmp.def` file, specifically pointing the source code to the **support/v2023.1** directory. After doing this, everything worked as expected. 

All `2023.2*znver2_psmp.def` and `2023.2*znver3_psmp.def` under `tools/apptainer` failed to build the container.

**Comment by mkrack** [comment]
Thanks for reporting that issue. It should be fixed by #2958 in the CP2K development version (master branch). You can try to apply the corresponding [patch](https://github.com/cp2k/cp2k/pull/2958/commits/bc29680dab0eb85bb5daa1ccef76ca866c3dd1af) to your local version. I would appreciate your feedback on whether the patch works.

**Comment by iamid0** [comment]
Fixed.
